psi4/src/psi4/libfock/dft_integrators.h

```diff
@@ -0,0 +1,331 @@
+    double** V2p = V->pointer();
+
+    // => LSDA contribution (symmetrized) <= //
+    double* v_rho_a = fworker->value("V_RHO_A")->pointer();
+    for (int P = 0; P < npoints; P++) {
+        std::fill(Tp[P], Tp[P] + nlocal, 0.0);
+        C_DAXPY(nlocal, 0.5 * v_rho_a[P] * w[P], phi[P], 1, Tp[P], 1);
+    }
+    // parallel_timer_off("LSDA Phi_tmp", rank);
+
+    // => GGA contribution (symmetrized) <= //
+    if (ansatz >= 1) {
+        // parallel_timer_on("GGA Phi_tmp", rank);
+        double** phix = pworker->basis_value("PHI_X")->pointer();
+        double** phiy = pworker->basis_value("PHI_Y")->pointer();
+        double** phiz = pworker->basis_value("PHI_Z")->pointer();
+        double* rho_ax = pworker->point_value("RHO_AX")->pointer();
+        double* rho_ay = pworker->point_value("RHO_AY")->pointer();
+        double* rho_az = pworker->point_value("RHO_AZ")->pointer();
+        double* v_sigma_aa = fworker->value("V_GAMMA_AA")->pointer();
+
+        for (int P = 0; P < npoints; P++) {
+            C_DAXPY(nlocal, w[P] * (2.0 * v_sigma_aa[P] * rho_ax[P]), phix[P], 1, Tp[P], 1);
+            C_DAXPY(nlocal, w[P] * (2.0 * v_sigma_aa[P] * rho_ay[P]), phiy[P], 1, Tp[P], 1);
+            C_DAXPY(nlocal, w[P] * (2.0 * v_sigma_aa[P] * rho_az[P]), phiz[P], 1, Tp[P], 1);
+        }
+        // parallel_timer_off("GGA Phi_tmp", rank);
+    }
+
+    // Collect V terms
+    C_DGEMM('T', 'N', nlocal, nlocal, npoints, 1.0, phi[0], coll_funcs, Tp[0], max_functions, 0.0, V2p[0],
+            max_functions);
+
+    for (int m = 0; m < nlocal; m++) {
+        for (int n = 0; n <= m; n++) {
+            V2p[m][n] = V2p[n][m] = V2p[m][n] + V2p[n][m];
+        }
+    }
+
+    // => Meta contribution <= //
+    if (ansatz >= 2) {
+        // parallel_timer_on("Meta", rank);
+        double** phix = pworker->basis_value("PHI_X")->pointer();
+        double** phiy = pworker->basis_value("PHI_Y")->pointer();
+        double** phiz = pworker->basis_value("PHI_Z")->pointer();
+        double* v_tau_a = fworker->value("V_TAU_A")->pointer();
+
+        double** phi_w[3];
+        phi_w[0] = phix;
+        phi_w[1] = phiy;
+        phi_w[2] = phiz;
+
+        for (int i = 0; i < 3; i++) {
+            double** phiw = phi_w[i];
+            for (int P = 0; P < npoints; P++) {
+                std::fill(Tp[P], Tp[P] + nlocal, 0.0);
+                C_DAXPY(nlocal, v_tau_a[P] * w[P], phiw[P], 1, Tp[P], 1);
+            }
+            C_DGEMM('T', 'N', nlocal, nlocal, npoints, 1.0, phiw[0], coll_funcs, Tp[0], max_functions, 1.0, V2p[0],
+                    max_functions);
+        }
+        // parallel_timer_off("Meta", rank);
+    }
+}
+
+inline void rks_gradient_integrator(std::shared_ptr<BasisSet> primary, std::shared_ptr<BlockOPoints> block,
+                                    std::shared_ptr<SuperFunctional> fworker, std::shared_ptr<PointFunctions> pworker,
+                                    SharedMatrix G, SharedMatrix U, int ansatz = -1) {
```

Comment:
this been `c-f`ed? may as well do the file from the start.

Reply:
Yes, this was `c-f`ed.